lib/components/app-service-bot.js

```diff
@@ -50,48 +56,37 @@ AppServiceBot.prototype.getJoinedRooms = function() {
  * Get a map of joined user IDs for the given room ID. The values in the map are objects
  * with a 'display_name' and 'avatar_url' properties. These properties may be null.
  * @param {string} roomId The room to get a list of joined user IDs in.
+ * @param {boolean=false} storeMembership Store membership found in this call in the bridge cache.
```

Comment:
tbh I'm having troubles justifying tagging on a caching behaviour to these functions, which is more what I was trying to get at with splitting out the caching layer (sorry). What's the use case for this?

Reply:
Probably the most obvious thing is that this is returning the know joined state for users in a room, so it would be smart to store it in the cache that is used to get the joined state for users in a room.

Otherwise we end up where we are today with intents pointlessly joining users to rooms.

Reply:
We could also ask the bridges to implement this themselves, but that seems kinda silly since caching is kinda internal. And a friendly Boolean is somewhat nicer than asking developers to loop round the result and store it in a cache object they now need to keep track of.

Reply:
Can we not just use the cache they now (optionally) provide? I don't think we need the flag unless there's ever a case where you'd request membership and not want it to be fast.

Reply:
I'm failing to see how this would be slow?

Reply:
By nature, not caching is slower than caching. So is there ever a use case where you wouldn't want to cache?